{{cookiecutter.repo_name}}/{{cookiecutter.repo_name}}/templates/forms/field.jinja

```diff
@@ -2,31 +2,67 @@
 {# or Sometimes third party controls come with their own styling, so for these widgets do not style #}
 {%-  macro input_widget_hook(field) -%}
     {%- if field.widget_type == "select"  -%}
-        select
+        input-box select
+    {%- elif field.widget_type == "radioselect" -%}
+        radio
     {%- else -%}
-        input
+        input-box input
     {%- endif -%}
 {%- endmacro -%}
 
-{% macro field(field, errors='', layout_classes='w-full') %}
-    <div class="{{ layout_classes }} {{ field.css_classes() }}">
-        <div class="field {% if errors %} error{% endif %}">
-            <div class="input-box {{ input_widget_hook(field) -}}">
-            {{ field }}
-            {% if field.label %}{{ field.label_tag() }}{% endif %}
-            {% if errors  %}
-                <div class="right-icon">
-                    {{heroicon_solid("exclamation-circle", class="text-error")}}
-                </div>
-            {% endif %}
+{% macro input_shell(field, errors='') %}
+    <div class="field {% if errors %} error{% endif %}">
+        <div class="input-box {{ input_widget_hook(field) }}">
+        {{ field }}
+        {% if field.label %}{{ field.label_tag() }}{% endif %}
+        {% if errors %}
+            <div class="right-icon">
+                {{heroicon_solid("exclamation-circle", class="text-error")}}
+            </div>
+        {% endif %}
+        </div>
+    </div>
+{% endmacro %}
+
+{% macro boolean_shell(field, errors='') %}
+    <div class="boolean-field flex items-center {% if errors %}error{% endif %}">
+        {% if field.label %}{{ field.label_tag() }}{% endif %}
+        {{ field }}
+    </div>
+{% endmacro %}
+
+{% macro radio_shell(field, errors='') %}
+    <div class="field {% if errors %} error{% endif %}">
+        <div class="input-box {{ input_widget_hook(field) }}">
+        {{ field }}
+        {% if errors %}
+            <div class="right-icon">
+                {{heroicon_solid("exclamation-circle", class="text-error")}}
             </div>
+        {% endif %}
         </div>
+    </div>
+{% endmacro %}
+
+{% macro field(field, errors='', layout_classes='w-full') %}
+    {% set wrapper_element = "fieldset" if field.widget_type == "radioselect" else "div" %}
+    <{{ wrapper_element }} class="{{ layout_classes }} {{ field.css_classes() }}">
+        {% if field.widget_type == "toggle" %}
+            {{boolean_shell(field, errors)}}
+        {% elif field.widget_type == "radioselect" %}
+            <legend>{{ field.label }}</legend>
+            {{radio_shell(field, errors)}}
+        {% else %}
+            {{input_shell(field, errors)}}
+        {% endif %}
         {% if errors %}
-            {{ errors }}
+            <div class="text-error">
```

Comment:
My thought is that the shell should include the error, vs handled at the field

Reply:
currently, the error/help text layout is the same for all fields. if we moved it into the shell we'd be be duplicating it at the moment